### PR TITLE
GUI

### PR DESCRIPTION
**Pull request by vnmsklnk**

### CODE REVIEW

**Comment by vnmsklnk** [comment]
Выполнил второе задание из 11 домашки.

File: src/ArithmGUI/Views/MainWindow.axaml.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                    }
+                    return "Process finished!\n" + resString;
+                }
+                catch (Exception ex)
+                {
+                    return ex.Message;
+                }
+            });
+            task.ContinueWith(x => 
+                Dispatcher.UIThread.Post(() =>
```
Comment (gsvgit) [inline src/ArithmGUI/Views/MainWindow.axaml.cs]: А я правильно понимаю, что Вы сперва накапливаете весь результат, а потом разом его печатаете? А можно ли сделать так, чтобы печаталось по мере появления новых результатов?

File: src/ArithmGUI/Views/MainWindow.axaml.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        }
+        
+        public async void Open(object sender, RoutedEventArgs e)
+        {
+            var dialog = new OpenFileDialog();
+            dialog.Filters.Add(new FileDialogFilter { Extensions = { "txt" } });
+            var path = await dialog.ShowAsync(this);
+            if (path is not {Length: > 0}) return;
+            _codeBox.Text = await File.ReadAllTextAsync(path[0]);
+            _openedFilePath = path[0];
```
Comment (gsvgit) [inline src/ArithmGUI/Views/MainWindow.axaml.cs]: Ммм. Кажется, если я открою файл, то потеряю всё, что писал до этого. И мне даже ничего об этом не скажут.

File: src/ArithmGUI/Views/MainWindow.axaml.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                InitialFileName = _openedFilePath
+            };
+            var path = await dialog.ShowAsync(this);
+            if (path == null) return;
+            await File.WriteAllTextAsync(path, _codeBox.Text);
+            _openedFilePath = path;
+        }
+        
+        public void New(object sender, RoutedEventArgs e)
+        {
```
Comment (gsvgit) [inline src/ArithmGUI/Views/MainWindow.axaml.cs:148]: И при создании нового тоже всё потеряю.

**Review by gsvgit — CHANGES_REQUESTED** [review]

**Comment by gsvgit** [comment]
А где у Вас скрипт для локальной сборки всего этого добра?

**Comment by gsvgit** [comment]
Скрипт сборки падает:
```
./build.sh 
Tool 'fake-cli' (version '5.20.4') was restored. Available commands: fake
Tool 'paket' (version '6.0.7') was restored. Available commands: paket
Tool 'fornax' (version '0.13.1') was restored. Available commands: fornax

Restore was successful.
The last restore is still up to date. Nothing left to do.
run Default
Building project with version: LocalBuild
Shortened DependencyGraph for Target Default:
<== Default
   <== Test
      <== Build
         <== Clean

The running order is:
Group - 1
  - Clean
Group - 2
  - Build
Group - 3
  - Test
Group - 4
  - Default
Starting target 'Clean'
Finished (Success) 'Clean' in 00:00:00.0053116
Starting target 'Build'
Starting task 'DotNet:build'
> "/usr/bin/mono" --version (In: false, Out: true, Err: true)
/home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW> "dotnet"  msbuild /version /nologo (In: false, Out: true, Err: true)
16.11.1.47101
/home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW> "dotnet"  build "" --configuration Release /bl:/tmp/tmpNXY1uD.tmp.binlog (In: false, Out: false, Err: false)
Microsoft (R) Build Engine version 16.11.1+3e40a09f8 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

/usr/share/dotnet/sdk/5.0.402/MSBuild.dll -consoleloggerparameters:Summary -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,/usr/share/dotnet/sdk/5.0.402/dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,/usr/share/dotnet/sdk/5.0.402/dotnet.dll -maxcpucount -property:Configuration=Release -restore -verbosity:m /bl:/tmp/tmpNXY1uD.tmp.binlog ./SecondSemester.sln
  Determining projects to restore...
/home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW/src/ArithmGUI/ArithmGUI.csproj : error NU1101: Unable to find package LongArithm. No packages exist with this id in source(s): nuget.org [/home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW/SecondSemester.sln]
  Failed to restore /home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW/src/ArithmGUI/ArithmGUI.csproj (in 1.09 sec).
  6 of 7 projects are up-to-date for restore.

Build FAILED.

/home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW/src/ArithmGUI/ArithmGUI.csproj : error NU1101: Unable to find package LongArithm. No packages exist with this id in source(s): nuget.org [/home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW/SecondSemester.sln]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:03.88
RestoreTask: /home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW/src/ArithmGUI/ArithmGUI.csproj(0,0): error NU1101: Unable to find package LongArithm. No packages exist with this id in source(s): nuget.org
Finished (Failed) 'DotNet:build' in 00:00:05.0706263
Finished (Failed) 'Build' in 00:00:05.0813742

---------------------------------------------------------------------
Build Time Report
---------------------------------------------------------------------
Target     Duration
------     --------
Clean      00:00:00.0028050
Build      00:00:05.0812592   (Exception of type 'Fake.DotNet.MSBuildException' was thrown.)
Test       00:00:00           (skipped)
Default    00:00:00           (skipped)
Total:     00:00:05.1969619
Status:    Failure
---------------------------------------------------------------------
Script reported an error:
-> BuildFailedException: Target 'Build' failed.
-> One or more errors occurred. (Exception of type 'Fake.DotNet.MSBuildException' was thrown.)
-> MSBuildException: Exception of type 'Fake.DotNet.MSBuildException' was thrown.
Hint: To further diagnose the problem you can run fake in verbose mode `fake -v run ...` or set the 'FAKE_DETAILED_ERRORS' environment variable to 'true'
Hint: Could not find a version in your paket.dependencies file, consider adding 'version 5.257.0' at the top of your dependencies file (/home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW/paket.dependencies).
Read https://github.com/fsharp/FAKE/issues/2193 for details.
Hint: The fake-runner has not been updated for at least 6 months. Please consider upgrading to get latest bugfixes, improved suggestions and F# features.
Performance:
 - Cli parsing: 219 milliseconds
 - Packages: 59 milliseconds
 - Script analyzing: 42 milliseconds
 - Script running: 5 seconds
 - Script cleanup: 122 milliseconds
 - Runtime: 6 seconds

```

**Comment by vnmsklnk** [comment]
> Скрипт сборки падает:
> 
> ```
> ./build.sh 
> Tool 'fake-cli' (version '5.20.4') was restored. Available commands: fake
> Tool 'paket' (version '6.0.7') was restored. Available commands: paket
> Tool 'fornax' (version '0.13.1') was restored. Available commands: fornax
> 
> Restore was successful.
> The last restore is still up to date. Nothing left to do.
> run Default
> Building project with version: LocalBuild
> Shortened DependencyGraph for Target Default:
> <== Default
>    <== Test
>       <== Build
>          <== Clean
> 
> The running order is:
> Group - 1
>   - Clean
> Group - 2
>   - Build
> Group - 3
>   - Test
> Group - 4
>   - Default
> Starting target 'Clean'
> Finished (Success) 'Clean' in 00:00:00.0053116
> Starting target 'Build'
> Starting task 'DotNet:build'
> > "/usr/bin/mono" --version (In: false, Out: true, Err: true)
> /home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW> "dotnet"  msbuild /version /nologo (In: false, Out: true, Err: true)
> 16.11.1.47101
> /home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW> "dotnet"  build "" --configuration Release /bl:/tmp/tmpNXY1uD.tmp.binlog (In: false, Out: false, Err: false)
> Microsoft (R) Build Engine version 16.11.1+3e40a09f8 for .NET
> Copyright (C) Microsoft Corporation. All rights reserved.
> 
> /usr/share/dotnet/sdk/5.0.402/MSBuild.dll -consoleloggerparameters:Summary -distributedlogger:Microsoft.DotNet.Tools.MSBuild.MSBuildLogger,/usr/share/dotnet/sdk/5.0.402/dotnet.dll*Microsoft.DotNet.Tools.MSBuild.MSBuildForwardingLogger,/usr/share/dotnet/sdk/5.0.402/dotnet.dll -maxcpucount -property:Configuration=Release -restore -verbosity:m /bl:/tmp/tmpNXY1uD.tmp.binlog ./SecondSemester.sln
>   Determining projects to restore...
> /home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW/src/ArithmGUI/ArithmGUI.csproj : error NU1101: Unable to find package LongArithm. No packages exist with this id in source(s): nuget.org [/home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW/SecondSemester.sln]
>   Failed to restore /home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW/src/ArithmGUI/ArithmGUI.csproj (in 1.09 sec).
>   6 of 7 projects are up-to-date for restore.
> 
> Build FAILED.
> 
> /home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW/src/ArithmGUI/ArithmGUI.csproj : error NU1101: Unable to find package LongArithm. No packages exist with this id in source(s): nuget.org [/home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW/SecondSemester.sln]
>     0 Warning(s)
>     1 Error(s)
> 
> Time Elapsed 00:00:03.88
> RestoreTask: /home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW/src/ArithmGUI/ArithmGUI.csproj(0,0): error NU1101: Unable to find package LongArithm. No packages exist with this id in source(s): nuget.org
> Finished (Failed) 'DotNet:build' in 00:00:05.0706263
> Finished (Failed) 'Build' in 00:00:05.0813742
> 
> ---------------------------------------------------------------------
> Build Time Report
> ---------------------------------------------------------------------
> Target     Duration
> ------     --------
> Clean      00:00:00.0028050
> Build      00:00:05.0812592   (Exception of type 'Fake.DotNet.MSBuildException' was thrown.)
> Test       00:00:00           (skipped)
> Default    00:00:00           (skipped)
> Total:     00:00:05.1969619
> Status:    Failure
> ---------------------------------------------------------------------
> Script reported an error:
> -> BuildFailedException: Target 'Build' failed.
> -> One or more errors occurred. (Exception of type 'Fake.DotNet.MSBuildException' was thrown.)
> -> MSBuildException: Exception of type 'Fake.DotNet.MSBuildException' was thrown.
> Hint: To further diagnose the problem you can run fake in verbose mode `fake -v run ...` or set the 'FAKE_DETAILED_ERRORS' environment variable to 'true'
> Hint: Could not find a version in your paket.dependencies file, consider adding 'version 5.257.0' at the top of your dependencies file (/home/gsv/Projects/IDEs_2sem_2021/Moskalenko/II-Semester-HW/paket.dependencies).
> Read https://github.com/fsharp/FAKE/issues/2193 for details.
> Hint: The fake-runner has not been updated for at least 6 months. Please consider upgrading to get latest bugfixes, improved suggestions and F# features.
> Performance:
>  - Cli parsing: 219 milliseconds
>  - Packages: 59 milliseconds
>  - Script analyzing: 42 milliseconds
>  - Script running: 5 seconds
>  - Script cleanup: 122 milliseconds
>  - Runtime: 6 seconds
> ```

Для успешного выполнения билд-скрипта необходимо добавить github source моего напарника. Сделать это можно такой командой:

dotnet nuget add source --username <YOUR_USERNAME> --password <PAT> --store-password-in-clear-text --name github "https://nuget.pkg.github.com/m0rphed/index.json"

**Comment by gsvgit** [comment]
Открываю IDE. Пишу код:
```
x = 3

print (x)

x = 1111111
y = 2222222

z = x ^ y

print (z)
```
Нажимаю "Start". В консоль выдаётся 
```
Processing...
3
Process finished
```
что, очевидно, не правда, так как ещё должен быть напечатан ```z```.

**Comment by m0rphed** [comment]
Кажется это проблема парсера / интерпретатора :disappointed:

**Review by gsvgit — APPROVED** [review]
Зачтено.

**Pull request merged**